### PR TITLE
Fix #1436. Change HPO term web links.

### LINKED ISSUE #1436: Kan inte nå HPO fr SCOUT
Eller ligger den servern nere bara???

### PR DESCRIPTION
**Pull request by dnil**
This PR fixes a bug/change in external API.

The Charité HPO service have begun forwarding to hpo.jax.org, but not to the correct place in their REST API yet. Changing to links directly to hpo.jax.org. Monthly dumps and phenomizer are still available on Charité.

As a bonus, add rank model links.
**How to test**:
1. click HPO term links (on case, phenotype and clinvar pages)
1. Click a (recent) rank model version (on the case sidebar).

**Expected outcome**:
The functionality should be working.

**Review:**
- [x] code approved by @moonso 
- [x] tests executed by @moonso and @travis

### CODE REVIEW

**Comment by dnil** [comment]
Ready for review.

File: scout/constants/case_tags.py [hunk 1]
```diff
@@ -51,3 +51,8 @@
 }
 
 VERBS = list(VERBS_MAP.keys())
+
+RANK_MODEL_LINK_PREFIX = "https://github.com/Clinical-Genomics/reference-files/blob/master/rare-disease/rank_model/rank_model_-v"
```
[the remark this answers is not in the document]
Comment (dnil) [reply scout/constants/case_tags.py]: That is a good question. If we put it in config, we will have to update the server config. I'm fine either way, but it does call for some kind of documentation, doesn't it?

[the remark this answers is not in the document]
Comment (moonso) [reply scout/constants/case_tags.py]: Yes but I think that sounds like a good idea. Just a small docs line in the server config. I can do it

File: scout/server/blueprints/cases/controllers.py [hunk 2]
```diff
@@ -137,9 +136,19 @@ def case(store, institute_obj, case_obj):
     case_obj['default_genes'] = list(distinct_genes)
     for hpo_term in itertools.chain(case_obj.get('phenotype_groups', []),
                                     case_obj.get('phenotype_terms', [])):
-        hpo_term['hpo_link'] = ("http://compbio.charite.de/hpoweb/showterm?id={}"
+        hpo_term['hpo_link'] = ("http://hpo.jax.org/app/browse/term/{}"
                                 .format(hpo_term['phenotype_id']))
 
+    rank_model_link_prefix = current_app.config.get('RANK_MODEL_LINK_PREFIX')
+    if case_obj.get('rank_model_version')and rank_model_link_prefix:
```
Comment (dnil) [inline scout/server/blueprints/cases/controllers.py]:
```suggestion
    if case_obj.get('rank_model_version'):
```

File: scout/server/blueprints/cases/controllers.py [hunk 3]
```diff
@@ -137,9 +136,19 @@ def case(store, institute_obj, case_obj):
     case_obj['default_genes'] = list(distinct_genes)
     for hpo_term in itertools.chain(case_obj.get('phenotype_groups', []),
                                     case_obj.get('phenotype_terms', [])):
-        hpo_term['hpo_link'] = ("http://compbio.charite.de/hpoweb/showterm?id={}"
+        hpo_term['hpo_link'] = ("http://hpo.jax.org/app/browse/term/{}"
                                 .format(hpo_term['phenotype_id']))
 
+    rank_model_link_prefix = current_app.config.get('RANK_MODEL_LINK_PREFIX')
```
Comment (dnil) [inline scout/server/blueprints/cases/controllers.py]:
```suggestion
    rank_model_link_prefix = current_app.config.get('RANK_MODEL_LINK_PREFIX','')
```

File: scout/server/blueprints/cases/controllers.py [hunk 4]
```diff
@@ -137,9 +136,19 @@ def case(store, institute_obj, case_obj):
...
+    if case_obj.get('rank_model_version')and rank_model_link_prefix:
+        rank_model_link_postfix = current_app.config.get('RANK_MODEL_LINK_POSTFIX','')
+        case_obj['rank_model_link'] = str(rank_model_link_prefix +
+                                      case_obj['rank_model_version'] + rank_model_link_postfix)
+    sv_rank_model_link_prefix = current_app.config.get('SV_RANK_MODEL_LINK_PREFIX')
```
Comment (dnil) [inline scout/server/blueprints/cases/controllers.py]:
```suggestion
    sv_rank_model_link_prefix = current_app.config.get('SV_RANK_MODEL_LINK_PREFIX','')
```

File: tests/server/blueprints/cases/test_cases_controllers.py [hunk 5]
```diff
@@ -48,7 +50,31 @@ def test_case_controller(adapter):
     assert data['found_cases'] == 1
 
 
-def test_cases_controller(adapter, institute_obj):
+def test_case_controller_rank_model_link(adapter, institute_obj):
```
Comment (dnil) [inline tests/server/blueprints/cases/test_cases_controllers.py]: :+1:

**Pull request merged**